app/code/Magento/Newsletter/Model/ResourceModel/Subscriber.php

```diff
@@ -157,6 +181,52 @@ public function loadByCustomerData(\Magento\Customer\Api\Data\CustomerInterface
         return [];
     }
 
+    /**
+     * Load subscriber by customer
+     *
+     * @param \Magento\Customer\Api\Data\CustomerInterface $customer
+     * @return array
+     */
+    public function loadByCustomerDataAndStore(\Magento\Customer\Api\Data\CustomerInterface $customer)
+    {
+        $select = $this->connection
+            ->select()->from($this->getMainTable())
+            ->where('customer_id=:customer_id and store_id=:store_id');
```

Comment:
Those two queries could be combined into one `WHERE (customer_id=:customer_id OR subscriber_email=:subscriber_email) AND store_id=:store_id`